src/main/java/org/testng/TestNG.java

```diff
@@ -275,23 +275,28 @@ public void initializeSuitesAndJarFile() {
 
     m_isInitialized = true;
     if (m_suites.size() > 0) {
-    	//to parse the suite files (<suite-file>), if any
-    	for (XmlSuite s: m_suites) {
-        for (String suiteFile : s.getSuiteFiles()) {
-            Path rootPath = Paths.get(s.getFileName()).getParent();
-            try {
-                Collection<XmlSuite> childSuites = getParser(rootPath.resolve(suiteFile).normalize().toString()).parse();
-                for (XmlSuite cSuite : childSuites){
-                    cSuite.setParentSuite(s);
-                    s.getChildSuites().add(cSuite);
-                }
-            } catch (ParserConfigurationException | IOException | SAXException e) {
-                e.printStackTrace(System.out);
-            }
-        }
-
-    	}
-      return;
+	    //to parse the suite files (<suite-file>), if any
+		for (XmlSuite s: m_suites) {
+		   for (String suiteFile : s.getSuiteFiles()) {
+	        	Path suitePath = Paths.get(s.getFileName());
+	        	Path rootPath = suitePath.getParent();
+	        	try {
+	        		Collection<XmlSuite> childSuites = getParser(rootPath.resolve(suiteFile).normalize().toString()).parse();
+	        		for (XmlSuite cSuite : childSuites){
+	        			Path childSuite = Paths.get(cSuite.getFileName()).normalize();
+	        			Path parentSuite = Paths.get(suiteFile).normalize();
+	        			if(!childSuite.getFileName().equals(parentSuite.getFileName()))
```

Comment:
There are two things here
1. parse() has a contract which says it will parse the current suite and also any children that it might have. So its perfectly fine for parse method to return this
2. Its a public method and will be kind of a change that will impact all who have used this way of running tests. Unless we are trying to add an overload of XmlSuite.parse

This was my original thinking to fix the problem. I dont think that we should try to change much if issue can be fixed with a surgical and logical code change.

If you still think that what you have suggested should be done, please explain more to me about your approach may be with some example.